pkg/tsdb/prometheus/ensure_http_method_middleware.go

```diff
@@ -0,0 +1,57 @@
+package prometheus
+
+import (
+	"net/http"
+	"strings"
+
+	sdkhttpclient "github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+	"github.com/grafana/grafana/pkg/infra/log"
+)
+
+func ensureHttpMethodMiddleware(logger log.Logger) sdkhttpclient.Middleware {
+	return sdkhttpclient.NamedMiddlewareFunc("prom-ensure-http-method", func(opts sdkhttpclient.Options, next http.RoundTripper) http.RoundTripper {
+		// this middleware only does something if the httpMethod setting is set to GET
+		grafanaData, exists := opts.CustomOptions["grafanaData"]
+		if !exists {
+			return next
+		}
+
+		data, ok := grafanaData.(map[string]interface{})
+		if !ok {
+			return next
+		}
+
+		methodInterface, exists := data["httpMethod"]
+		if !exists {
+			return next
+		}
+
+		method, ok := methodInterface.(string)
+		if !ok {
+			return next
+		}
```

Comment:
Nit. Rather than doing this here you could take `httpMethod` as an argument to ensureHttpMethodMiddleware and get the `httpMethod` from settings in `newInstanceSettings`.`customParamsMiddleware` could probably use the same approach.

Reply:
hi @marefr thanks for the tip. i adjusted the code this way:
- the middleware always forces the GET_http_method
- we only add the middleware when the JSON says so

could you please check it again, if it is ok? my only worry is:
is the code going to re-initialize correctly when the datasource-json changes?

based on my tests, it does work correctly,  i'm only asking to be sure. thanks!